### PR TITLE
Internal modules

### PR DESCRIPTION
**Pull request by Guria**
preparing to extract some internals as modules
cc @christianalfoni

### CODE REVIEW

**Comment by Guria** [comment]
Woohooo.
@christianalfoni I have freed signal factory from custom logic. please review it. I will continue to cleanup devtools, signalStore and recorder for now.

File: src/CreateSignalFactory.js [hunk 1]
```diff
@@ -66,37 +52,38 @@ module.exports = function (signalStore, recorder, devtools, controller, model, s
...
         }
 
-        if (!signalStore.isRemembering() && !recorder.isCatchingUp()) {
+        if (!options.branches) {
           controller.emit('signalStart', {signal: signal})
         }
 
-        if (recorder.isRecording()) {
-          recorder.addSignal(recorderSignal)
+        if (signal.isPrevented) {
```
Comment (christianalfoni) [inline src/CreateSignalFactory.js]: Hm, how can this become true? :)

Comment (christianalfoni) [inline src/CreateSignalFactory.js]: Ah, you mean emitting event and preventDefualt on it?

**Comment by Guria** [comment]
I want leave my notes here, for the history :)

File: src/CreateSignalFactory.js (hunk 1 above)
Comment (Guria) [inline src/CreateSignalFactory.js]: yeap. you can run `signal.preventSignalRun()` in `signalStart` event callback.

**Comment by bfitch** [comment]
Wow. Awesome work @Guria!

File: src/CreateSignalFactory.js (hunk 1 above)
Comment (christianalfoni) [inline src/CreateSignalFactory.js]: :+1:

**Comment by Guria** [comment]
@christianalfoni ready for review. I think it is time to merge. I'll do it after your LGTM comment

**Comment by Guria** [comment]
@christianalfoni maybe you want to try to release it on your own? If you are confident:
```sh
git checkout master
git pull master # check that your local branch is up to date with origin/master
git merge internal-modules # should be a fast-forward
git push origin master # github wouldn't reject it since all tests on this pr is green
```

**Pull request merged**